### ISSUE TITLE
terraform always wants to recreate azurerm_api_management_api_tag

### ISSUE THREAD

**Issue by sven5** [hashicorp/terraform-provider-azurerm#24705]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.1

### AzureRM Provider Version

3.89.0

### Affected Resource(s)/Data Source(s)

azurerm_api_management_api_tag

### Terraform Configuration Files

Hi,

When dealing with `azurerm_api_management_api_tag` terraform always wants to recreate the same item because the suffix  `;rev=10` is being ignored.

Example:

### Debug Output/Panic Output

Output from `tf plan` 

### module.api_management_2[0].azurerm_api_management_api_tag.apim_api_tag[11] must be replaced
```
-/+ resource "azurerm_api_management_api_tag" "apim_api_tag" {
      ~ api_id = "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.ApiManagement/service/apim001/apis/atest-2-v7" -> "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.ApiManagement/service/apim001/apis/test-2-v7;rev=10" # forces replacement
      ~ id     = "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.ApiManagement/service/apim001/apis/test-2-v7/tags/test2" -> (known after apply)
        name   = "test2"
    }
```

After applying, the same change is still present when issuing a new plan.

Output from `tf state` for this resource:

### module.api_management_2[0].azurerm_api_management_api_tag.apim_api_tag[11]:
```
resource "azurerm_api_management_api_tag" "apim_api_tag" {
    api_id = "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.ApiManagement/service/apim001/apis/test-2-v7"
    id     = "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.ApiManagement/service/apim001/apis/test-2-v7/tags/test2"
    name   = "test2"
}

```

We can see that the suffix `;rev=10` is missing in the state.

### Expected Behaviour

The API tag should not be identified as a change because it's already present.

### Actual Behaviour

The API tag is always recreated.

**Comment by alexwiese** [comment]
Looks like this, and a few other issues, were introduced by #23031

**Comment by Aujjani** [comment]
Hello @sven5 @alexwiese i am still facing this issue .
is there any fix ?

in my case with each plan 45 tags get replaced so its a big issue for me .
please help

**Comment by sven5** [comment]
I haven't been working on our terraform scripts for a while, but I assume this still exists.

**Comment by alexwiese** [comment]
Hi @Aujjani 

You can workaround the issue by stripping the suffix off of the tag ID before passing it as an input to the resources. 

`api_id = split(";", azurerm_api_management_api.example.id)[0]`

The [PR that fixes this ](#25168) is still waiting to be merged by the maintainers.

**Comment by sven5** [comment]
We're still waiting for the merge... 😢

**Comment by stephybun** [comment]
@J0F3 there is a detailed discussion within the PR on the issue as well the fix. 

TL;DR we can't roll the fix out until we release the next major version 4.0 of the provider since all of the tags need to be recreated. This has been implemented behind a feature flag so the functionality is in the provider, but won't be switched on until 4.0.

**Comment by J0F3** [comment]
@stephybun Ah ok, got you. Thanks for the update.
But then I still think that this issue should be kept open until the problem is really fixed and the fix is available for everyone. Otherwise I find it very confusing and in additional the chances are high that the problem get reported again with other issues because no open issue can be found anymore.
